crates/bevy_ecs/src/component.rs

```diff
@@ -12,26 +12,85 @@ use std::{
     any::{Any, TypeId},
 };
 
-/// A component is data associated with an [`Entity`](crate::entity::Entity). Each entity can have
-/// multiple different types of components, but only one of them per type.
+// TODO: The content in the section “Implementing the trait for foreign types” may be more appropriate
+// for the Bevy book. However, until the change gets published, it's better to keep it there. Once the
+// book includes usage of the pattern, the section can be replaced with a link to the appropriate
+// section of the book.
+
+/// A data type that can be associated to an [entity].
```

Comment:
```suggestion
/// A data type that can be used to store data for an [entity].
```
Your existing phrasing is very precise, but it's ultimately unapproachable for beginners.